protocol/synthetix/contracts/interfaces/IVaultModule.sol

```diff
@@ -56,14 +164,181 @@ interface IVaultModule {
      *
      * Emits a {DelegationUpdated} event.
      */
-    function delegateCollateral(
```

Comment:
Is it possible to have the interfaces set up where atomic delegation can still be possible and the interface doesn't change? Like this function could be the "process" function if necessary a delay is set?

There's also an abstraction where everything is async, and atomic can be done with a batch of two tx's when the delay is configured to 0.

Reply:
i belive switching the delay and window to 0 leads to atomic processing with the same interface

Reply:
i have same feedback